### ISSUE TITLE
twisted.words.protocol.irc.IRC.whois references undefined variable RPL_WHOISOPER,

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2089]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)| @jerub reported|
|-|-|
|Trac ID|trac#2089|
|Type|defect|
|Created|2006-09-14 04:02:40Z|

RPL_WHOISOPER should be RPL_WHOISOPERATOR I think.

<details><summary>Searchable metadata</summary>

```
trac-id__2089 2089
type__defect defect
reporter__jerub jerub
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1158206560000000 1158206560000000
changetime__1158903767000000 1158903767000000
version__None None
owner__moonfallen moonfallen
cc__corydodt@...
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_1"></a>|@jerub set owner to moonfallen|
|-|-|

```
Index: protocols/irc.py
===================================================================
--- protocols/irc.py    (revision 18110)
+++ protocols/irc.py    (working copy)
@@ -423,7 +423,7 @@
             self.hostname, RPL_WHOISSERVER, user, nick, server, serverInfo))
         if oper:
             self.sendLine(":%s %s %s %s :is an IRC operator" % (
-                self.hostname, RPL_WHOISOPER, user, nick))
+                self.hostname, RPL_WHOISOPERATOR, user, nick))
         self.sendLine(":%s %s %s %s %d %d :seconds idle, signon time" % (
             self.hostname, RPL_WHOISIDLE, user, nick, idle, signOn))
         self.sendLine(":%s %s %s %s :%s" % (
```

Test needed.

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|moonfallen removed owner|
|-|-|

Created branch irc-whois-2089 and added a unit test for whois.  This is ready for review now.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml<br>@jml set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml set owner to moonfallen<br>@jml set status to `new`|
|-|-|

Looks good to me. Please add a docstring to `testWhois`, then merge.

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|moonfallen set status to `closed`|
|-|-|

(In [18224]) Author: moonfallen, jerub
Reviewer: jml
Fixes [#2089](https://github.com/twisted/twisted/issues/2089)

Use the correct name for the constant and a test for whois.
